### PR TITLE
Use spacy model to compute semantic similarity

### LINKED ISSUE #56: Implement better similarity metric for selecting DWD overlay
Currently, we are using Dice Coefficient to get the most similar Q node. We should switch this one based on semantic similarity like SentenceBERT.

### PR DESCRIPTION
**Pull request by elizlee**
Closes #56 

Comparison on test documents vs. string similarity:
[string_sim_vs_spacy_sem_sim.xlsx](https://github.com/joecummings/cdse-covid/files/7590313/string_sim_vs_spacy_sem_sim.xlsx)

### CODE REVIEW

**Comment by joecummings** [comment]
Based on the test document you produced, it seems like the SpaCy model does work slightly better!

File: amr-requirements-lock.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ dataclasses==0.6
 decorator==5.1.0
 deprecation==2.1.0
 dill==0.2.9
-en-core-web-sm @ https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-2.0.0/en_core_web_sm-2.0.0.tar.gz
+en-core-web-md @ https://github.com/explosion/spacy-models/releases/download/en_core_web_md-2.0.0/en_core_web_md-2.0.0.tar.gz
```
Comment (joecummings) [inline amr-requirements-lock.txt:18]: Why the switch to the larger model? Did you find worse results with the small one?

Comment (elizlee) [reply amr-requirements-lock.txt:18]: The `sm` models don't ship with word vectors (https://v2.spacy.io/usage/vectors-similarity#_title). Similarity can still be calculated using context-sensitive tensors, but the results were a bit less accurate when I tried it. The `lg` model has more unique vectors, but it seems like `md` works well enough for our purposes. Though, it should be easy enough to switch to the larger version if we want.

Comment (joecummings) [reply amr-requirements-lock.txt:18]: Makes sense - cool.

**Review by joecummings — APPROVED** [review]
lgtm

**Pull request merged**